### PR TITLE
[SPARK-42315][SQL] Assign name to _LEGACY_ERROR_TEMP_(2091|2092)

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to assign name to _LEGACY_ERROR_TEMP_2091 and _LEGACY_ERROR_TEMP_2091, "CANNOT_READ_FOOTER".

### Why are the changes needed?

We should assign proper name to _LEGACY_ERROR_TEMP_*

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

`./build/sbt "sql/testOnly org.apache.spark.sql.SQLQueryTestSuite*`

### CODE REVIEW

**Comment by itholic** [comment]
cc @srielau @MaxGekk @cloud-fan Please take a look at this error class review ticket when you find some time 🙏

**Review by srielau — APPROVED** [review]

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -93,6 +93,11 @@
     ],
     "sqlState" : "22007"
   },
+  "CANNOT_READ_FOOTER" : {
```
Comment (srielau) [inline core/src/main/resources/error/error-classes.json:96]:
```suggestion
  "CANNOT_READ_FILE_FOOTER" : {
```

Comment (itholic) [reply core/src/main/resources/error/error-classes.json:96]: Thanks!

**Comment by MaxGekk** [comment]
@itholic Could you rebase on the recent master, please.

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master/3.4.
Thank you, @itholic and @srielau for review.

**Pull request closed without merging**